### PR TITLE
feat(frontend): display swap status in a single toast

### PR DESCRIPTION
**Pull request by boriskubrik**
- Render a single toast and update it's state to communicate the status of the swap. This is the MVP version of the Swap Modal (#453). The toast is unclosable until the swap is completed. Uses `getJump` to track the status of jumps.
- Update balances when transactions are executed and completed. Refactored logic to `useRefetchBalances` hook.
- Update the execute button's label to show "Execute Swap" or "Execute Jump" [91139f6](https://github.com/sifiorg/sifi/pull/503/commits/91139f63d1b9d287c28020d4b46c0b4ee485b1d2)
- Adjust space travel animation so it is synced with the jump. Decided to slow down during "arriving", as it is fairly long and we don't want users to get an epileptic seizure.

### Future plans
1. Communicate what swap is being executed. Perhaps show icons as part of the loading animation. This requires updates to Toast in shared-ui.
2. #453

### Testing
- [x] Swap works as expected
- [x] Jump works as expected
- [x] Looks good on mobile in MetaMask
- [x] Balances update after sending and receiving
- [x] Toast can only be closed when close icon is visible

### Video

_Edited out about 3 minutes of "Arriving" state._

https://github.com/sifiorg/sifi/assets/128667801/6d8d6ea3-9750-4f42-91ac-8f3d2cedabe1

### CODE REVIEW

File: packages/frontend/src/hooks/useSwapToast.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+      autoClose: false,
+    });
+
+    await publicClient.waitForTransactionReceipt({ hash });
+    refetchAllBalances();
+
+    // TODO: What if the transaction fails?
+
+    if (isJump) {
+      const intervalId = setInterval(async () => {
```
Comment (immutablegoblin) [inline packages/frontend/src/hooks/useSwapToast.tsx:47]: This could probably be done in a cleaner manner with react-query, not essential

Comment (tommy0x) [reply packages/frontend/src/hooks/useSwapToast.tsx:47]: Yes. Could we do the polling through React Query, and update the toast through a useEffect, or through a react-query callback if one is available?

**Comment by xykota** [comment]
Looks great! Loving the role-playing text

**Review by immutablegoblin — APPROVED** [review]
Working nicely - one comment on a possible refactor, but happy to approve regardless

File: packages/frontend/src/hooks/useSwapToast.tsx (hunk 1 above)
Comment (boriskubrik) [reply packages/frontend/src/hooks/useSwapToast.tsx:47]: We can do this refactor later.

**Pull request merged**